### PR TITLE
Add aws migrator skeleton

### PR DESCRIPTION
**Pull request by calvix**
add skeleton for aws migrator, most copy-pasted from azure
+ add  flag to run migratory based on provider,  keeping azure as default, for now, to be backward compatible

based on `update-crs` branch to make the diff  less noisy

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -140,22 +143,36 @@ func mainE(ctx context.Context) error {
...
 	{
-		azureMigratorFactory, err = migration.NewAzureMigratorFactory(migration.AzureMigrationConfig{
-			CtrlClient:    mgr.GetClient(),
-			Logger:        log,
-			TenantCluster: tenantCluster,
-		})
-		if err != nil {
-			return microerror.Mask(err)
+		if flags.Provider == "aws" {
+			migratorFactory, err = migration.NewAzureMigratorFactory(migration.AzureMigrationConfig{
```
Comment (whites11) [inline main.go:149]: I guess you want to initialize AWSMigratoFactory here?

Comment (calvix) [reply main.go:149]: yeah you are right , nice catch

File: go.mod [hunk 2]
```diff
@@ -4,6 +4,7 @@ go 1.16
 
 require (
 	github.com/Azure/go-autorest/autorest/to v0.4.0
+	github.com/giantswarm/apiextensions v0.4.20
```
Comment (whites11) [inline go.mod:7]: why do we need such an old version of apiexternsions?

File: main.go [hunk 3]
```diff
@@ -56,11 +56,14 @@ func init() {
 var flags = struct {
 	EnableLeaderElection bool
 	MetricsAddr          string
+	Provider             string
 }{}
 
 func initFlags() {
 	flag.BoolVar(&flags.EnableLeaderElection, "leader-elect", false, "Enable leader election for controller manager.")
 	flag.StringVar(&flags.MetricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&flags.Provider, "provider", "azure", "Provider name for the migration.")
```
Comment (whites11) [inline main.go:65]: Why don't we make the provider flag required and avoid a default?
I appreciate the fact you wanted to keep backward compatibility but I think it's best to have the flag fully implemented from the beginning

File: go.mod (hunk 2 above)
Comment (calvix) [reply go.mod]: yeah noticed that too, just wrong package imports

**Review by whites11 — APPROVED** [review]
LGTM

File: main.go (hunk 3 above)
Comment (calvix) [reply main.go:65]: If you are fine with that Than I am more than happy to do it, just did not want to break your  testing

Comment (whites11) [reply main.go]: go for it, thx

**Pull request merged**